src/mdns/mdns_avahi.cpp

```diff
@@ -519,6 +519,11 @@ bool PublisherAvahi::IsStarted(void) const
 
 void PublisherAvahi::Stop(void)
 {
+    VerifyOrExit(mState == State::kReady);
+
+    mState = State::kIdle;
+    mStateCallback(mState);
```

Comment:
should this callback be called after everything is cleared?

Reply:
See discussion here: https://github.com/openthread/ot-br-posix/pull/2677#discussion_r1958727572